From ab9fb85b0bb54810b3548c6fa8ece7a94cf544e3 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Wed, 17 Feb 2021 14:58:20 +0800
Subject: [PATCH] Revert "usb: gadget: composite: fix dead lock when uvc
 disconnect"

This reverts commit b43b8174c1cddfd68b38684825e385b186802681.

Relpaced by commit fc9f57f643c3 ("usb: gadget: Fix spinlock lockup on usb_function_deactivate")

Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/usb/gadget/composite.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index 4288dae23dda..7d1fb7f893c7 100644
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -392,12 +392,8 @@ int usb_function_deactivate(struct usb_function *function)
 
 	spin_lock_irqsave(&cdev->lock, flags);
 
-	if (cdev->deactivations == 0) {
-		/* drop the spinlock so the gadget disconnect can run */
-		spin_unlock_irqrestore(&cdev->lock, flags);
+	if (cdev->deactivations == 0)
 		status = usb_gadget_deactivate(cdev->gadget);
-		spin_lock_irqsave(&cdev->lock, flags);
-	}
 	if (status == 0)
 		cdev->deactivations++;
 
-- 
2.35.3

